src/spyglass/decoding/decoding_merge.py

```diff
@@ -84,18 +84,36 @@ def cleanup(self, dry_run=False):
                     except (PermissionError, FileNotFoundError):
                         logger.warning(f"Unable to remove {path}, skipping")
 
-    @classmethod
-    def _get_source_class(cls, key):
-        if cls._source_class_dict is None:
-            cls._source_class_dict = {}
-            module = inspect.getmodule(cls)
-            for part_name in cls.parts():
+    @property
+    def source_class_dict(self) -> dict:
+        """Dictionary of source class names to source classes
+
+        {
+            'ClusterlessDecodingV1': spy...ClusterlessDecodingV1,
+             'SortedSpikesDecodingV1': spy...SortedSpikesDecodingV1
+        }
+
+        Returns
+        -------
+        dict
+            Dictionary of source class names to source classes
+        """
+        if not self._source_class_dict:
+            self._ensure_dependencies_loaded()
+            module = inspect.getmodule(self)
+            for part_name in self.parts():
```

Comment:
In a previous PR to this branch, my intent was to say 'we could do a or b'. Sorry I wasn't explicit. A was making this decoding merge table responsible for the class dict, and B was to make the Merge class responsible. It looks like both are preserved in this PR to master. If the Merge class version works for your purposes, I would use that so other tables can benefit

In this case, `load_result` and other methods would `return cls().merge_get_parent_class(key)`

Reply:
Yeah, I think I just haven't had time to rectify these. Could you maybe PR to this branch with that?